system/lib/libcxxabi/src/cxa_exception.cpp

```diff
@@ -280,6 +287,14 @@ __cxa_throw(void *thrown_object, std::type_info *tinfo, void (*dest)(void *)) {
 
 #ifdef __USING_SJLJ_EXCEPTIONS__
     _Unwind_SjLj_RaiseException(&exception_header->unwindHeader);
+#elif __USING_WASM_EXCEPTIONS__
+#ifdef NDEBUG
+    _Unwind_RaiseException(&exception_header->unwindHeader);
+#else
+    // In debug mode, call a JS library function to use WebAssembly.Exception JS
+    // API, which enables us to include stack traces
+    __throwCppWebAssemblyException(&exception_header->unwindHeader, true);
```

Comment:
What about calling this `_Unwind_RaiseException_WithStacktrace`?   Since its basically the same function as `_Unwind_RaiseException`?   Maybe it doesn't need a second argument if its always called with true?

Reply:
`_Unwind_` functions are the naming convention of libunwind, and this function is in `library_exceptions.js`.. So I named it like this. On the other hand, `library_exception.js`'s (or other library JS files') naming schemes are kind of mixed... Some are camelCase, some are snake_case, and some are with two underscores, some are with one underscore, some are without... So it was kind of not very easy to name it. Should it have leading underscores or not?

The reason I added the boolean option is in case users want to directly use that function to throw `WebAssembly.Exception` from JS, in which case they may or may not want the stack traces.

Reply:
But we don't have such users yet do we?  So maybe we can keep the code simple for now and make it more complex later if it proves useful, and also keep this function private/internal.

Also, a user who doesn't want a stacktrace could/should presumably just call the existing `_Unwind_RaiseException` function.

Reply:
OK, will remove the argument. Not sure how to rename the function (given that `_Unwind` is libunwind prefix and this is not libunwind), will leave this as is for now. Please let me know if you have a better idea.

Reply:
Oh, will add `WithStackTrace`, now that the arg is gone.

Reply:
Hmm, given that we decided to use this as an internal function for the moment, will rename this to `__throw_exception_with_stack_trace`. (Not sure about JS library naming schemes, but it looks internal functions are likely to be in snake_case...) Let me know if you have a better idea.